log-manager/src/main/java/io/airlift/log/BufferedHandler.java

```diff
@@ -143,43 +146,25 @@ private void logging()
 
     private void processQueue()
     {
-        List<byte[]> batch = new ArrayList<>(MAX_BATCH_COUNT);
-        boolean poisonMessageSeen = false;
-        while (!closed.get() || !poisonMessageSeen) {
-            if (queue.isEmpty()) {
-                try {
-                    batch.add(queue.take());
-                }
-                catch (InterruptedException ignored) {
-                }
-            }
-            else {
-                queue.drainTo(batch, MAX_BATCH_COUNT);
+        byte[] message = new byte[0];
+        while (!closed.get()) {
+            try {
+                message = queue.poll(100, TimeUnit.MILLISECONDS);
             }
-
-            int poisonMessageIndex = getPoisonMessageIndex(batch);
-            if (poisonMessageIndex >= 0) {
-                poisonMessageSeen = true;
-                batch = batch.subList(0, poisonMessageIndex);
+            catch (InterruptedException ignored) {
             }
 
-            logMessageBatch(batch);
-            batch.clear();
-        }
-    }
-
-    private synchronized void logMessageBatch(List<byte[]> batch)
-    {
-        for (byte[] message : batch) {
-            try {
-                messageOutput.writeMessage(message);
-            }
-            catch (Exception e) {
-                reportError(null, e, WRITE_FAILURE);
+            if (message != POISON_MESSAGE && message != null) {
```

Comment:
It seems like there is a difference here where before we were stopping when we saw the poison message, this doesn't seem to stop, just skip when it encounters the poison message..